### PR TITLE
Add storage classes for supported SSD types

### PR DESCRIPTION
**Pull request by baixiac**

### CODE REVIEW

File: config/ebs.tf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    apiVersion: storage.k8s.io/v1 
+    kind: StorageClass
+    metadata:
+      name: ${local.storage_classes.gp2}
+    provisioner: ebs.csi.aws.com
+    volumeBindingMode: WaitForFirstConsumer
+    allowVolumeExpansion: true
+    reclaimPolicy: Retain
+    parameters:
+      type: gp3
```
Comment (K1Hyve) [inline config/ebs.tf:12]: Should be gp2?

**Comment by K1Hyve** [comment]
Is it possible define one class as default one during the setup and avoid touching the Kubernetes configuration after the setup?

**Comment by baixiac** [comment]
That would be nice but at the time of writing it looks not possible since neither the csi-driver addon nor the eks module supports that. The only way I found is to change the default post cluster creation, sth like:
```bash
kubectl patch storageclass gp2 -p '{"metadata": {"annotations":{"storageclass.beta.kubernetes.io/is-default-class":"false"}}}'
kubectl patch storageclass radar-base-ebs-sc-gp3 -p '{"metadata": {"annotations":{"storageclass.beta.kubernetes.io/is-default-class":"false"}}}
```

File: config/ebs.tf (hunk 1 above)
Comment (baixiac) [reply config/ebs.tf]: Well spotted. Rectified.

**Comment by K1Hyve** [comment]
I've seen people running Helm with Terraform so I think it should be possible to run Kubernetes commands with Terraform as well.

**Comment by baixiac** [comment]
Alright, with `kubernetes_annotations` resources, the default eks gp2 is now disabled and `radar-base-ebs-sc-gp2` is set as default instead coz it is more cost-effective than other custom storage classes.

**Review by K1Hyve — APPROVED** [review]
Thanks, LGTM! 
Would be good if other options were also configurable but I guess they're not too important for now.

**Comment by baixiac** [comment]
Definitely. Also free feel to make it more customisable based off your use cases.

**Pull request merged**